File: code/dsa/linked_lists/javascript/zip_iterative.js
function zipLinkedList(head1, head2) {
    let tail = head1;
    let current1 = head1;
    let current2 = head2;
    let count = 0

    while ((current1 !== null) && (current2 !== null)) {
        if (count % 2 === 0) {
            tail.next = current2;
            current2 = current2.next;
        } else {
            tail.next = current1;
            current1 = current1.next;
        }
        tail = tail.next;
        count += 1;
    }
    if (current1 !== null) {
        tail.next = current1;
    }
    if (current2 !== null) {
        tail.next = current2;
    }
    return head1
}

File: source/code/dsa/linked_lists/python/zip_iterative.py
def zip_linked_lists(head1, head2):
    """Zips two linked lists together."""
    tail = head1
    current1 = head1
    current2 = head2
    count = 0
    while (current1 is not None) and (current2 is not None):
        if count % 2 == 0:
            tail.next = current2
            current2 = current2.next
        else:
            tail.next = current1
            current1 = current1.next
        tail = tail.next
        count += 1
    if current1 is not None:
        tail.next = current1
    if current2 is not None:
        tail.next = current2
    return head1
